### PR TITLE
Add expand/collapse to all sections of dottydoc

### PR DESCRIPTION
**Pull request by Varunram**
Intended as a fix for #2211. This is a preliminary version (there's a lot of code to be deleted regarding the short-body part of Constructors/Members). Please do comment your views :)

### CODE REVIEW

File: doc-tool/resources/css/api-page.css [hunk 1]
```diff
@@ -142,3 +159,7 @@ div.entity-section > div.member > div.member-body > div.member-body-long {
 div.entity-section > div.member > div.member-body > div.member-body-long.toggled {
     display: block;
 }
+
+div.entity-section.toggled{
```
Comment (felixmulder) [inline doc-tool/resources/css/api-page.css]: Space after `toggled`

File: doc-tool/resources/_layouts/api-page.html [hunk 2]
```diff
@@ -169,21 +171,27 @@ <h1 class="section {% if entity.hasVisibleMembers == false %}empty{% endif %}">
...
+                            var members = document.getElementById("entity-constructors");
+                            members.classList.toggle("toggled");
+                            span.innerHTML = (span.innerHTML == "[-]") ?  "[+]" : "[-]";
+                        }
+
+                        function toggleMembersBody(span, sig){
+                            var members = document.getElementById("entity-members");
+                            members.classList.toggle("toggled");
+                            span.innerHTML = (span.innerHTML == "[-]") ?  "[+]" : "[-]";
+                        }
```
Comment (felixmulder) [inline doc-tool/resources/_layouts/api-page.html]: This should be the same function - but perhaps taking the `members` variable as argument.

Comment (felixmulder) [inline doc-tool/resources/_layouts/api-page.html]: I.e. `toggleMembersBody` and `toggleConstructorsBody`

File: doc-tool/resources/css/api-page.css (hunk 1 above)
Comment (Varunram) [inline doc-tool/resources/css/api-page.css]: Will do :)

File: doc-tool/resources/_layouts/api-page.html (hunk 2 above)
Comment (Varunram) [inline doc-tool/resources/_layouts/api-page.html]: Got it! Will fix in the next commit. Is the edit with ```entities.scala``` acceptable?

Comment (felixmulder) [inline doc-tool/resources/_layouts/api-page.html]: Yes - and it will be deleted soon anyway, right?

Comment (Varunram) [inline doc-tool/resources/_layouts/api-page.html]: Yeah, along with other stuff. Wanted to confirm if with the deletion.

**Comment by Varunram** [comment]
@felixmulder Removed instances of comment.short and changed a few CSS class names. Please comment your reviews and I'll fix the commit history to ```Fix #2211: ...```. Thanks!

File: doc-tool/resources/_layouts/rand.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+currdate=$(date "+%Y-%m-%d\ %H:%M:%S")
+currDateComment="<!-- Docs Last Generated On "$currdate"-->"
+sed -i '.original' "1s/^/$currDateComment\\
+/" main.html
```
Comment (felixmulder) [inline doc-tool/resources/_layouts/rand.sh]: This script should probably not be included in the PR

**Comment by felixmulder** [comment]
Not by a computer currently, will test this tomorrow when I'm back at work, I want to see how it looks. Code looks fine otherwise! :+1:

File: doc-tool/resources/_layouts/rand.sh (hunk 3 above)
Comment (Varunram) [inline doc-tool/resources/_layouts/rand.sh]: Yeah Sorry! Will fix it when I fix the commit history. Thanks!

**Comment by Varunram** [comment]
@felixmulder I think you'd have to revert that, courtesy rand.sh

**Comment by felixmulder** [comment]
@Varunram - don't worry about it. Already took care of it in #2487 :)

**Comment by Varunram** [comment]
I was waiting for your final call so I'd fix up the commit messages and stuff. Sorry

**Comment by felixmulder** [comment]
No worries, I squashed away the nastiness - this shouldn't be a full time job for you - you're a contributor :)
